src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Services/Ingestion/PDFDirectorySource.cs

```diff
@@ -52,12 +52,12 @@ public async Task<IEnumerable<SemanticSearchRecord>> CreateRecordsForDocumentAsy
     {
         using var pdf = PdfDocument.Open(Path.Combine(sourceDirectory, documentId));
         var paragraphs = pdf.GetPages().SelectMany(GetPageParagraphs).ToList();
-        
+
         var embeddings = await embeddingGenerator.GenerateAsync(paragraphs.Select(c => c.Text));
 
         return paragraphs.Zip(embeddings).Select((pair, index) => new SemanticSearchRecord
         {
-            Key = $"{Path.GetFileNameWithoutExtension(documentId)}_{pair.First.PageNumber}_{pair.First.IndexOnPage}",
+            Key = Guid.CreateVersion7(),
```

Comment:
Qdrant only support `Guid` and `long` keys. I know the original code generated deterministic keys based on paragraph data. If that's important (I'm not sure whether it is), we can change this line.

Reply:
Update: Azure AI Search only support `string` keys... so we'll need to figure out what to do here. We _could_ do some preprocessor magic to generate code using the correct key type depending on the vector database being used by the template. But then what if the developer wants to change their code to use a different vector database? Another option is to adjust the template-defined abstractions to be more agnostic to the key type.

Reply:
Going with some preprocessing logic for now.